internal/testutil/acc/config_formatter_test.go

```diff
@@ -104,3 +108,257 @@ func TestFormatToHCLLifecycleIgnore(t *testing.T) {
+  project_id     = "project"
+
+  replication_specs {
+    num_shards = 1
+    zone_name  = "Zone 1"
+
+    region_configs {
+      priority      = 7
+      provider_name = "AWS"
+      region_name   = "US_WEST_1"
+      auto_scaling {
+        disk_gb_enabled = false
+      }
+      electable_specs {
+        instance_size = "M10"
+        node_count    = 3
+      }
+    }
+
+    region_configs {
+      priority      = 7
+      provider_name = "AWS"
+      region_name   = "EU_WEST_1"
+      auto_scaling {
+        disk_gb_enabled = false
+      }
+      electable_specs {
+        instance_size = "M10"
+        node_count    = 3
+      }
+    }
+  }
+
+}
+`
+
+func Test_ClusterResourceHcl(t *testing.T) {
+	var (
+		clusterName                 = "my-name"
+		replicationSpecMultiRegions = admin.ReplicationSpec{
+			NumShards: conversion.IntPtr(1),
+			ZoneName:  conversion.StringPtr("Zone 1"),
+			RegionConfigs: &[]admin.CloudRegionConfig{
+				acc.CloudRegionConfig("US_WEST_1", constant.AWS, "M10", 3),
+				acc.CloudRegionConfig("EU_WEST_1", constant.AWS, "M10", 3),
+			},
+		}
+		testCases = map[string]struct {
+			expected string
+			req      acc.ClusterRequest
+		}{
+			"defaults": {
+				standardClusterResource,
+				acc.ClusterRequest{ClusterNameExplicit: clusterName},
+			},
+			"dependsOn": {
+				dependsOnClusterResource,
+				acc.ClusterRequest{ClusterNameExplicit: clusterName, ResourceDependencyName: "mongodbatlas_project.project_execution"},
+			},
+			"dependsOnMulti": {
+				dependsOnMultiResource,
+				acc.ClusterRequest{ClusterNameExplicit: clusterName, ResourceDependencyName: fmt.Sprintf("%s, %s", "mongodbatlas_private_endpoint_regional_mode.atlasrm", "mongodbatlas_privatelink_endpoint_service.atlasple")},
+			},
+			"twoReplicationSpecs": {
+				twoReplicationSpecs,
+				acc.ClusterRequest{ClusterNameExplicit: clusterName, ReplicationSpecs: []admin.ReplicationSpec{
+					acc.ReplicationSpec(&acc.ReplicationSpecRequest{Region: "US_WEST_1", ZoneName: "Zone 1"}),
+					acc.ReplicationSpec(&acc.ReplicationSpecRequest{Region: "EU_WEST_2", ZoneName: "Zone 2"}),
+				}},
+			},
+			"overrideClusterResource": {
+				overrideClusterResource,
+				acc.ClusterRequest{ClusterNameExplicit: clusterName, Geosharded: true, CloudBackup: true, ReplicationSpecs: []admin.ReplicationSpec{
+					acc.ReplicationSpec(&acc.ReplicationSpecRequest{
+						Region: "MY_REGION_1", ZoneName: "Zone X", InstanceSize: "M30", NodeCount: 30, ProviderName: "AZURE",
+					}),
+				}},
+			},
+			"twoRegionConfigs": {
+				twoRegionConfigs,
+				acc.ClusterRequest{ClusterNameExplicit: clusterName, ReplicationSpecs: []admin.ReplicationSpec{replicationSpecMultiRegions}},
+			},
+		}
+	)
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			config, actualClusterName, err := acc.ClusterResourceHcl("project", &tc.req)
+			require.NoError(t, err)
+			assert.Equal(t, clusterName, actualClusterName)
+			assert.Equal(t, tc.expected, config)
```

Comment:
knit: we're checking exactly the same string, i.e. not that they're equivalent HCL files, for instance they can have different indentations and newlines but represent the same HCL.  don't know if there is a native HCL comparison function. 

the problem with this is that we might want to change the Prod code (imagine indentation) and tests will fail and will have to be adapted even if behavior is kept.   tests are "over specified".

Reply:
Seems like they are "over specifying" in the [hcl repo too](https://github.com/hashicorp/hcl/blob/360ae579460fab69e9939599af85c8f255a59007/hclwrite/examples_test.go#L14)
An alternative is tokens, but almost impossible to specify those manually: https://github.com/hashicorp/hcl/blob/main/hclwrite/format_test.go#L793